### DISCUSSION TITLE
Is it possible to display files on cloud storage

### DISCUSSION THREAD

**Discussion by logica-academy** [jupyterlite/jupyterlite#1045] [Q&A]
Is it possible to display files on cloud storage in jupyterLite's file manager?
If so, how can this be accomplished?

**Comment by jtpio** [comment]
@logica-academy it *may* be possible to access files from a cloud storage if a JupyterLab extension provides a way to do so via a custom Drive. However some cloud storage providers require using tokens to access their API which can bring some challenges as to how they can be used in a JupyterLite deployment.

For example the `jupyterlab-github` extension (available on the https://jupyterlite.rtfd.io/en/latest/try/lab demo site) provides a `Drive` that enables accessing files from a public GitHub repo:

![image](https://user-images.githubusercontent.com/591645/231073858-f2114c73-8b4a-425a-9d87-de79723a539d.png)

**Reply by logica-academy** [reply]
@jtpio 
Thanks for the reply.

We are working to teach programming to children in Japan.
We believe that JupyterLite will be the best execution environment for children to learn data science.
So, we are planning to have each child open Jupyter Lite on the server and coding program.
What do you think is the best way for each child to save and open his or her program?

**Reply by jtpio** [reply]
This sounds like a great use case!

> on the server and coding program.

Just to be sure, what do you mean by "server"?

Do you already have a custom cloud storage and would like to interface with it from JupyterLite? If so then you will likely need to implement a JupyterLab extension to be able to fetch the files from a remote location.

Currently with JupyterLite `0.1.0` there could be two ways of doing this:

- implement a separate `Drive` as mentioned above. The drive would add itself to the JupyterLite interface in the left area, separate from the default file browser
- switch the default JupyterLite contents manager to use your own instead. This would need to be implemented as a JupyterLite "server" extension: https://jupyterlite.readthedocs.io/en/latest/howto/extensions/server.html

With JupyterLite `0.2.0` we are considering switching the default contents manager to be a regular drive: https://github.com/jupyterlite/jupyterlite/issues/1041. This could help streamline how the contents is managed, and site deployers could more easily choose which "backend" they want for storing the files.

**Reply by logica-academy** [reply]
Thanks for your reply.

Being able to save and open files to their own local drive would be very helpful so that students can manage their own files.

I look forward to them being able to use that soon.

As you have probably experienced, one of the biggest obstacles for learners when they start learning programming is setting up a development environment.

Thank you so much for creating JupyterLite.

**Reply by jtpio** [reply]
> Being able to save and open files to their own local drive would be very helpful so that students can manage their own files.

Maybe a workaround for now would be to use the `jupyterlab-filesystem-access` extension? https://jupyterlite.readthedocs.io/en/latest/howto/content/filesystem-access.html

It allows for accessing local files on the host machine, directly from the browser. Although it's only supported in Chromium based browsers for now.
